src/entrypoint.mjs

```diff
@@ -62,6 +63,14 @@ export default async function cli(options) {
     };
   }
 
+  // Configure "native" `fetch` (e.g. `globalThis.fetch` in Node.js)
+  // to work with the `global-agent` proxy
+  // @see: https://github.com/gajus/global-agent/issues/52#issuecomment-1134525621
+  const ProxyAgent = Undici.ProxyAgent;
+  const setGlobalDispatcher = Undici.setGlobalDispatcher;
+
+  setGlobalDispatcher(new ProxyAgent(info.proxy.http));
```

Comment:
The node script is launched in a different process (line 75) and the logic which does the interception is defined in the `constants.SCRIPT_HELPER` aka https://github.com/ballercat/jambox/blob/main/src/script-helper.js

Until you update _that_ script nothing will change with the actual spawned process, this just sets up an interceptor for the current jambox process not the actual server or whatever _you_ are running.

You can see how `httptoolkit` handles `undici` here https://github.com/httptoolkit/httptoolkit-server/blob/b74fdfb563bfea2d2c1ef35314e51c0f4b399abb/overrides/js/prepend-node.js

Reply:
Ah ha, figured I was doing something dumb 😂 moved these into the `script-helper.js` file made this start working as expected!

I assume Jambox doesn't actually load that `prepend-node` file? Is that one just for reference / if someone wasn't using `global-agent`?

Reply:
Jambox doesn't load it, node loads it for the script(s) you're trying to proxy. See the `--require` flag here https://github.com/ballercat/jambox/blob/cbadef83ba93d3c2051fc296c43926c3867378f5/src/entrypoint.mjs#L72

Node docs https://nodejs.org/api/cli.html#-r---require-module

> Preload the specified module at startup.

Reply:
Right, guess I'm wondering why we use a custom pre-require script instead of mockttp's one (I assume global-agent abstracts most of that for us, hence our own script that calls `global-agent`)